### PR TITLE
Men 5

### PR DESCRIPTION
**Pull request by jufglanville**
- Jira Ticket: [MEN-5](https://jufglanvlle.atlassian.net/browse/MEN-5?atlOrigin=eyJpIjoiZmJmYjA0MmFhNDBmNDQxOGFjYmNmZWU3MmFjMDViZjEiLCJwIjoiaiJ9)

- The PricePlan has been split into a new component. This component now accepts props for the `type` and `amount` with the user able to update between the two

### CODE REVIEW

File: src/components/PricePlan.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import icon from '../assets/icon-music.svg';
+
+interface Props {
+  type: 'annual' | 'monthly';
```
Comment (JakeStaTeresa) [inline src/components/PricePlan.tsx]: You could create a type alias for 'annual' | 'monthly', considering its being used in a few places.

File: src/components/SummaryCard.tsx [hunk 2]
```diff
@@ -1,10 +1,34 @@
 import React from 'react';
 import styled from 'styled-components';
 
-import icon from '../assets/icon-music.svg';
+import PricePlan from './PricePlan';
+
 import hero from '../assets/illustration-hero.svg';
 
+type PricePlanType = {
```
Comment (JakeStaTeresa) [inline src/components/SummaryCard.tsx]: Types could go under src/types

File: src/components/PricePlan.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+const PricePlan = ({ type, amount, handleClick }: Props) => {
+  return (
+    <Container>
+      <img src={icon} alt="Music Note" />
+      <PriceContainer>
+        <Heading>{type} Plan</Heading>
+        <Text>${amount}/year</Text>
+      </PriceContainer>
+      <ActionLink
+        onClick={() => handleClick(type === 'annual' ? 'monthly' : 'annual')}
```
Comment (JakeStaTeresa) [inline src/components/PricePlan.tsx]: Not sure if this ternary is correct.

File: src/components/SummaryCard.tsx [hunk 4]
```diff
@@ -1,10 +1,34 @@
...
+    type: 'monthly',
+    amount: 6.99,
+  },
+};
+
 const SummaryCard = () => {
+  const [selectedPlan, setSelectedPlan] = React.useState(pricePlan.annual);
+
   return (
     <Card>
```
Comment (JakeStaTeresa) [inline src/components/SummaryCard.tsx:35]: I think a PriceCard would be a suitable refactor. Let's say if we are going to create a Pricing page.

Comment (JakeStaTeresa) [reply src/components/SummaryCard.tsx:35]: We can loop through the available plans and show each one of them

**Review by JakeStaTeresa — CHANGES_REQUESTED** [review]

File: src/components/PricePlan.tsx (hunk 1 above)
Comment (jufglanville) [reply src/components/PricePlan.tsx]: I've added a new PriceType alias in src/types.ts file

File: src/components/SummaryCard.tsx (hunk 2 above)
Comment (jufglanville) [reply src/components/SummaryCard.tsx]: I've moved the other Types to this `src/types.ts` file and importing them when needed

**Comment by jufglanville** [comment]
I've added a PricePlanList component that loops over an array of all the available PricePlans and renders each of them. The `id` of the selected plan is then passed to the `SummaryCard` state.

File: src/components/PricePlan.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import styled from 'styled-components';
+import { PlanType, PriceType } from '../types';
+import icon from '../assets/icon-music.svg';
+
+interface Props {
+  plan: PlanType;
+  selected: boolean;
+  onHandleClick: (val: string) => void;
```
Comment (JakeStaTeresa) [inline src/components/PricePlan.tsx]: If the downstream variable is called selected, I would call the event propagating the variable onSelect. Its not that obvious but the reason is the as when you have a function called openFile, I would expect you to have another function called closeFile and not unsetFileHandle

File: src/components/PricePlan.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+type ButtonProps = {
```
Comment (JakeStaTeresa) [inline src/components/PricePlan.tsx]: Not sure what purpose this type serves here..

File: src/components/PricePlan.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  onClick: (event: React.MouseEventHandler<HTMLButtonElement>) => void;
+};
+
+const PricePlan = ({ plan, selected, onHandleClick }: Props) => {
+  return (
+    <Container>
+      <img src={icon} alt="Music Note" />
```
Comment (JakeStaTeresa) [inline src/components/PricePlan.tsx]: Nit: alt is hard coded

File: src/components/PricePlan.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+      <PriceContainer>
+        <Heading>{plan.type} Plan</Heading>
+        <Text>${plan.amount}/year</Text>
+      </PriceContainer>
+      {!selected && (
+        <ActionLink onClick={() => onHandleClick(plan.id)}>Change</ActionLink>
```
Comment (JakeStaTeresa) [inline src/components/PricePlan.tsx]: Nit: Change text is hard coded here. Might be a good a idea to allow the user to pass it through?

File: src/components/PricePlanList.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+import PricePlan from './PricePlan';
+import { PricePlans } from '../types';
+
+interface Props {
+  plans: PricePlans;
+  selected: string;
+  onHandleClick: (val: string) => void;
+}
+
+const PricePlanList = ({ plans, selected, onHandleClick }: Props) => {
```
Comment (JakeStaTeresa) [inline src/components/PricePlanList.tsx]: Ditto: onHandleClick should be onClick

**Review by JakeStaTeresa — CHANGES_REQUESTED** [review]

File: src/components/PricePlan.tsx (hunk 6 above)
Comment (jufglanville) [reply src/components/PricePlan.tsx]: It was to do with the StyledComponent, but it looks like it is redundant now so will remove

File: src/components/PricePlan.tsx (hunk 5 above)
Comment (jufglanville) [reply src/components/PricePlan.tsx]: By changing this it now makes a lot more sense in the SummaryCard component as this triggers the setSelectedPlan.

File: src/components/PricePlanList.tsx (hunk 9 above)
Comment (jufglanville) [reply src/components/PricePlanList.tsx]: I have updated this to also be onSelect, as the prop being passed is called selected. Would this make more sense?

File: src/components/PricePlan.tsx (hunk 8 above)
Comment (jufglanville) [reply src/components/PricePlan.tsx]: I've added a prop that is passed down from the SummaryCard component through the PricePlanList to the PricePlan component

**Comment by jufglanville** [comment]
I have done quite a big refactor in terms of the data structure and believe this to be more reusable. 
The `PricePlan` component now has no hard-coded parts, all the data is passed in as props, this is also the case for the `PricePlanList`.
The data is added in the `SummaryCard` component. Here I have a new object called `PricePlanTemplate` this object contains general data for the entire `PricePlan` elements:
- `actionLabel` - This is the label used in the `PricePlan` component that allows the user to switch between plans.
- `icon` - The SVG image used on each plan
- `altText` - The alt text used for the image
- `currency` - The currency used
- `plans` - This is the array of `PricePlans` themselves. 
This allows us to pass this single `PricePlanTemplate` object and get all the relevant data required.

**Comment by jufglanville** [comment]
I've updated the `PricePlanList` to pass the destructured properties, `actionLabel`, `icon`, `alText` and `currency` as opposed to passing the entire `template`.

File: src/components/PricePlanList.tsx (hunk 9 above)
Comment (JakeStaTeresa) [reply src/components/PricePlanList.tsx]: Indeed it does

**Pull request merged**